doc/api.md

```diff
@@ -1,128 +1,109 @@
 # IoT Agent API
 
-<!-- TOC WIP -->
```

Comment:
Maybe a first version of the ToC should be included in this PR, to see the new added topics.

Reply:
Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1336/commits/4af6d975efe94709b578ca45ca18cab90a0dc4e2